src/openzaak/components/documenten/models.py

```diff
@@ -498,6 +504,79 @@ def has_gebruiksrechten(self):
             return self.canonical.gebruiksrechten_set.exists()
 
 
+class BestandsDeel(models.Model):
+    uuid = models.UUIDField(
+        unique=True, default=_uuid.uuid4, help_text="Unieke resource identifier (UUID4)"
+    )
+    informatieobject = models.ForeignKey(
+        "EnkelvoudigInformatieObjectCanonical",
+        on_delete=models.CASCADE,
+        related_name="bestandsdelen",
+        null=True,
+        blank=True,
+    )
+    informatieobject_uuid = models.UUIDField(
+        verbose_name=_("EnkelvoudigInformatieObject UUID"),
+        help_text=_(
+            "De unieke identifier van het gerelateerde EnkelvoudigInformatieObject in het "
+            "achterliggende Document Management Systeem."
+        ),
+        blank=True,
+        null=True,
+    )
+    volgnummer = models.PositiveIntegerField(
+        help_text=_("Een volgnummer dat de volgorde van de bestandsdelen aangeeft.")
+    )
+    omvang = models.PositiveIntegerField(
+        help_text=_("De grootte van dit specifieke bestandsdeel.")
+    )
+    inhoud = PrivateMediaFileField(
+        upload_to="part-uploads/%Y/%m/",
+        blank=True,
+        help_text=_("De (binaire) bestandsinhoud van dit specifieke bestandsdeel."),
+    )
+    datetime_created = models.DateTimeField(_("datetime created"), default=timezone.now)
+
+    objects = BestandsDelenManager()
+
+    class Meta:
+        verbose_name = "bestands deel"
+        verbose_name_plural = "bestands delen"
+        unique_together = ("informatieobject", "volgnummer")
+        constraints = [
+            models.CheckConstraint(check=Q(omvang__gt=0), name="check_omvang",),
+        ]
```

Comment:
would add a unique together for `informatieobject_uuid` as well, and a check contraint that either `informatieboject` or `informatieobject_uuid` is filled out